lib/resolvers/parse/ast/literal.js

```diff
@@ -0,0 +1,57 @@
+const { Kind } = require('graphql')
+
+const _getTypeFrom_fieldOr_arg = _field => {
+  let { type } = _field
+  while (type.ofType) type = type.ofType
+  return type
+}
+
+const _getTypeFrom_fields = (_fields, path, index = 0) => {
+  const { name } = path[index++]
+
+  const _field = _fields[name]
+  const type = _getTypeFrom_fieldOr_arg(_field)
+
+  // If type has the parseLiteral function it is a scalar type -> leaf -> end of path
+  if (type.parseLiteral) return type
+
+  // Is the next path element an argument? If yes, follow the argument
+  if (path[index].kind === Kind.ARGUMENT) {
+    const arg = _field.args.find(a => a.name === path[index].name)
+    const type = _getTypeFrom_fieldOr_arg(arg)
+
+    // If type has the parseLiteral function it is a scalar type -> leaf -> end of path
+    // This case occurs when the argument itself is a scalar type, e.g. Books(top: 1)
+    if (type.parseLiteral) return type
+
+    return _getTypeFrom_fields(type.getFields(), path, index + 1)
+  }
+
+  return _getTypeFrom_fields(type.getFields(), path, index)
+}
+
+const _simplifiedPath = (node, ancestors) =>
+  ancestors
+    .concat(node)
+    .filter(e => [Kind.FIELD, Kind.ARGUMENT, Kind.OBJECT_FIELD].includes(e.kind))
+    .map(e => ({ kind: e.kind, name: e.name.value }))
```

Comment:
Make it clear to v8 that it is going to do the exact same thing many times.

```suggestion
const _pathKinds = [Kind.FIELD, Kind.ARGUMENT, Kind.OBJECT_FIELD]
const _isPathKind = _pathKinds.includes.bind(_pathKinds)
const _simplifiedPath = (node, ancestors) =>
  ancestors
    .concat(node)
    .filter(_isPathKind)
    .map(e => ({ kind: e.kind, name: e.name.value }))
```